CAIPs/caip-ipld_anchor_event.md

```diff
@@ -0,0 +1,150 @@
+
+Again IPLD offers natural and standard way to to encode a Merkle tree, with the added benefit of allowing anyone to easily resolve the entire tree, paths or subset of the tree when verifying a Merkle proof. We encode a IPLD Merkle tree as follows. 
+
+The set of data to be committed, or “leaves” of the Merkle tree are expected to be a set of IPLD CIDs. A binary Merkle tree is used, and a node value is constructed with two “leaves” or CIDs using a list. The data is encoded in IPLD with [dag-cbor](https://ipld.io/specs/codecs/dag-cbor/spec/), the resulting IPLD block results in another CID for this tree node. 
+
+IPLD Merke Tree Node Type
+
+```tsx
+type NodeList [Link]
+```
+
+Example 1
+
+```tsx
+[CID(bagcqcerav5zyrlelnnx24blbgob6stz6y4oiwenn3hodbmyf7bnnguh6gdeq), CID(bagcqcera4nnpzmwsk2bbww6n4bgq3cwne6gmhkouciam44nyjbindkxpz2eq)
+```
+
+Non leaf nodes are then recursively built in the same way using list pairs of CIDs until you have constructed the entire Merkle tree and a root Node/CID. 
+
+### IPLD Anchor Event
+
+An Anchor event is made up of both a Merkle tree root CID and the corresponding reference to a blockchain transaction which includes the Merkle tree root. The blockchain transaction can be referenced by a IPLD CID if there is a corresponding IPLD codec for the blockchain and the CAIP-2 chainID or it can be referenced by a transaction id string. 
+
+Note: There exists IPLD codecs for two of the most commonly used blockchains for anchoring, Bitcoin and Ethereum. An IPLD codec for a blockchain should not necessarily be required to represent an Anchor Event, so a transaction id string is supported as well. Transaction ids should have a corresponding CAIP before use, for now they are only referenced as an example and considered simple strings that any implementation must agree on or define in a future standard. 
+
+If there is `chainID` parameter, we expect the `txId` to be an IPLD link to the transaction encoded using the related blockchain IPLD codec.
+
+```tsx
+type AnchorEvent struct {
+  root Link
+	chainID String 
+	txId Link
+}
+```
+
+Example 2
+```tsx
+{
+  root: CID(bafyreiaxdhr5jbabn7enmbb5wpmnm3dwpncbqjcnoneoj22f3sii5vfg74)
+	chainID: "eip155:1"
+	txId: CID(bagjqcgzanbud4sqdsywfp2mckuj57qsffsovgyjhh7sxebkqwr335hzy2zbq)
+}
+```
+
+If there is not a `chainID` parameter, we expect `txId` to be a transaction id string.
+
+```tsx
+type AnchorEvent struct {
+  root Link
+	txId String
+}
+```
+
+Example 3
+```tsx
+{
+  root: CID(bafyreiaxdhr5jbabn7enmbb5wpmnm3dwpncbqjcnoneoj22f3sii5vfg74)
+	txId: "eip155:1/txid:0x58be3f7b9d508d12b0c1cf2015a54169598210f1b40bf645777bcaf9cae656e8"
+}
+```
```

Comment:
Why do we want to include this in the spec? Is this present in legacy Ceramic code?
I think we can just not mention it. Just an artifact of legacy Ceramic.

Reply:
I added it, no legacy, reference in note, if this is a general CAIP spec I cant see how there is not an option to reference any chain without requiring a ipld codec for it, plus to dont see a lot of benefit to needing a tx to be in ipld

Reply:
Ok, the benefit of referencing the tx by CID is twofold:

1. We don't need to specify any new CAIP. The txid string you suggest above would need a new CAIP
2. We can use IPLD paths to address the data, e.g. `<anchor-event-cid>/proof/txId/Data` to get the tx data

Reply:
i dont mind removing for now, just feels very limiting as it requires a ipld codec and for chain data to actually be published in ipfs/ipld (i.e you can not even resolve the eth tx data right now), its a lot of overhead for general chain support

spec has note on needing a new caip and not defining yet, unless someone wanted to use that, but similarly this can be removed, someone can add it if they want it in the future

Reply:
I don't see a problem adding the friction of having to add support for IPLD. Makes everything cleaner.
> you can not even resolve the eth tx data right now

Still think it's worth it for forwards compatibility. Laconic network will make this very viable.